neo.UnitTests/UT_PoolItem.cs

```diff
@@ -0,0 +1,113 @@
+                                              .Callback(() => timeIndex++);
+            TimeProvider.Current = timeMock.Object;
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            // important to leave TimeProvider correct
+            TimeProvider.ResetToDefault();
+        }
+
+        [TestMethod]
+        public void PoolItem_CompareTo_Fee()
+        {
+            int size1 = 50;
+            int netFeeSatoshi1 = 1;
+            var tx1 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshi1), size1, UInt256.Zero);
+            int size2 = 50;
+            int netFeeSatoshi2 = 2;
+            var tx2 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshi2), size2, UInt256.Zero);
+
+            PoolItem pitem1 = new PoolItem(tx1.Object);
+            PoolItem pitem2 = new PoolItem(tx2.Object);
+
+            Console.WriteLine($"item1 time {pitem1.Timestamp} item2 time {pitem2.Timestamp}");
+            // pitem1 < pitem2 (fee) => -1
+            pitem1.CompareTo(pitem2).Should().Be(-1);
+            // pitem2 > pitem1 (fee) => 1
+            pitem2.CompareTo(pitem1).Should().Be(1);
+        }
+
+        [TestMethod]
+        public void PoolItem_CompareTo_Hash()
+        {
+            int sizeFixed = 50;
+            int netFeeSatoshiFixed = 1;
+            var tx1 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshiFixed), sizeFixed, UInt256.Zero);
+            var tx2 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshiFixed), sizeFixed, new UInt256(TestUtils.GetByteArray(32, 0x42)));
+
+            PoolItem pitem1 = new PoolItem(tx1.Object);
+            PoolItem pitem2 = new PoolItem(tx2.Object);
+            // pitem2 < pitem1 (fee) => -1
+            pitem2.CompareTo(pitem1).Should().Be(0); // -1
+            // It's not passing hash to mock object yet! TODO
+        }
+
+        [TestMethod]
+        public void PoolItem_CompareTo_Equals()
+        {
+            int sizeFixed = 500;
+            int netFeeSatoshiFixed = 10;
+            var tx1 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshiFixed), sizeFixed, UInt256.Zero);
+            var tx2 = MockGenerateInvocationTx(new Fixed8(netFeeSatoshiFixed), sizeFixed, UInt256.Zero);
+
+            PoolItem pitem1 = new PoolItem(tx1.Object);
+            PoolItem pitem2 = new PoolItem(tx2.Object);
+            // pitem1 == pitem2 (fee) => 0
+            pitem1.CompareTo(pitem2).Should().Be(0);
+        }
+
+        // Generate Mock InvocationTransaction with different sizes and prices
+        public static Mock<Transaction> MockGenerateInvocationTx(Fixed8 networkFee, int size, UInt256 hash)
+        {
+            var mockTx = new Mock<Transaction>(TransactionType.InvocationTransaction);
+            mockTx.SetupGet(mr => mr.NetworkFee).Returns(networkFee);
+            mockTx.SetupGet(mr => mr.Size).Returns(size);
+
+            //mockTx.SetupGet(mr => mr.Hash).Returns(hash); // cannot overwrite this method, will see GetHashData
```

Comment:
You can look at how the MemoryPool does it. You need to set the parameters of the transaction that will be used when computing the hash. If you want to generate one with a lesser or greater hash you can generate multiple until you get one that meets your criteria.

Reply:
Implemented what I suggested here and pushed to this PR.